### PR TITLE
hepmc3: 3.1.2 -> 3.2.0, add python support

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build python27Packages.hepmc3 python37Packages.hepmc3 python38Packages.hepmc3

**Comment by veprbl** [comment]
@GrahamcOfBorg build python27Packages.hepmc3 python37Packages.hepmc3 python38Packages.hepmc3

File: pkgs/development/libraries/physics/hepmc3/default.nix [hunk 1]
```diff
@@ -1,28 +1,41 @@
-{ stdenv, fetchurl, cmake, coreutils, root }:
+{ stdenv, fetchurl, cmake, coreutils, python, root }:
+
+let
+  pythonVersion = with stdenv.lib.versions; "${major python.version}${minor python.version}";
```
Comment (jonringer) [inline pkgs/development/libraries/physics/hepmc3/default.nix]:
```suggestion
  pythonVersion = stdenv.lib.versions.majorMinor python.version";
```

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -2327,8 +2331,6 @@ in {
 
   pyhcl = callPackage ../development/python-modules/pyhcl { };
 
-  pyhepmc = callPackage ../development/python-modules/pyhepmc { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix]: at the very least add a deprecation mechanic:
```
pyhepmc = throw "please use hepmc3";
```

**Review by jonringer — COMMENTED** [review]
with how python is developed, I don't think saying "just use this other package" is enough (in regards to pyhepmc).

File: pkgs/development/libraries/physics/hepmc3/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/development/libraries/physics/hepmc3/default.nix]: There is an important difference:
```
nix-repl> with (import ./. {}); with stdenv.lib.versions; "${major python.version}${minor python.version}"
"27"

nix-repl> with (import ./. {}); stdenv.lib.versions.majorMinor python.version                             
"2.7"
```

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/top-level/python-packages.nix]: Thanks for reminding me about this. I've now considered this and I don't think this is needed in this particular case.

File: pkgs/development/libraries/physics/hepmc3/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/libraries/physics/hepmc3/default.nix]: ah nevermind then

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/top-level/python-packages.nix]: the other option would be to alias, but I don't like aliasing in python packages, they should reflect the name used on pypi (if available)

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM

```
python27Packages.bacpypes python37Packages.bacpypes python38Packages.bacpypes
```

**Pull request merged**